todo-angular-express/public/js/controllers/todoCtrl.js

```diff
@@ -103,24 +103,24 @@ todomvc.controller('TodoCtrl', function TodoCtrl($scope, $routeParams, todoStora
 	};
 
 	$scope.removeTodo = function (todo) {
-	    todoStorage.delete(todo.id).success(function() {
-            $scope.todos.splice($scope.todos.indexOf(todo), 1);
-        }).error(function() {
-            alert("Failed to delete this TODO");
-        });
+		todoStorage.delete(todo.id).success(function() {
+			$scope.todos.splice($scope.todos.indexOf(todo), 1);
+		}).error(function() {
+			alert('Failed to delete this TODO');
+		});
 	};
 
 	$scope.clearCompletedTodos = function () {
-		$scope.todos = todos.filter(function (val) {
+		$scope.todos = $scope.todos.filter(function (val) {
```

Comment:
I've noticed this as well, but I know nothing about angular (changing this line did at least get rid of the exception because `todos` was undefined). I'll have a look at it.